### PR TITLE
some small modifications

### PR DESCRIPTION
**Pull request by xiaoyu-dong**
Some corrections of typos,  adding a few sentences in some of the documents, and some minor reorders of the codes to make it more readable. No essential change of any functions. I make a pull request just to make it easier to compare, and you may want to accept some of them.

### CODE REVIEW

File: src/spaces/vectorspaces.jl [hunk 1]
```diff
@@ -67,11 +68,14 @@ function dim end
 """
     dual(V::VectorSpace) -> VectorSpace
 
-Return the dual space of `V`; also obtained via `V'`. It is assumed that
-`typeof(V) == typeof(V')`.
+Return the dual space of `V`; also obtained via `V'`. This should satisfy 
+dual(dual(V))==V. It is assumed that `typeof(V) == typeof(V')`.
```
Comment (Jutho) [inline src/spaces/vectorspaces.jl]: Could you make this
```
`dual(dual(V)) == V`
```
 i.e. with backticks and spaces. Then it prints more nicely as code. Thanks.

File: src/spaces/vectorspaces.jl [hunk 2]
```diff
@@ -163,7 +168,7 @@ function flip end
 """
     conj(V::S) where {S<:ElementarySpace} -> S
 
-Return the conjugate space of `V`.
+Return the conjugate space of `V`. This should satisfy conj(conj(V))==V.
```
Comment (Jutho) [inline src/spaces/vectorspaces.jl]: Same:
```
`conj(conj(V)) == V`
```

File: src/spaces/vectorspaces.jl [hunk 3]
```diff
@@ -172,32 +177,35 @@ Base.conj(V::ElementarySpace{ℝ}) = V
 """
     abstract type InnerProductSpace{𝕜} <: ElementarySpace{𝕜} end
 
-Abstract type for denoting vector with an inner product and a corresponding metric, which
-can be used to raise or lower indices of tensors.
+Abstract type for denoting vector with an inner product, thus a canonical mapping from
```
Comment (Jutho) [inline src/spaces/vectorspaces.jl]: This should have been: `... for denoting vector spaces with an inner product...`, i.e. i had forgotten the word spaces.

**Comment by Jutho** [comment]
Oh, and thanks for catching the `infimum` mistake. You probably want to search the whole package for instances of `infinum`, including the tests, and the exports in the file `TensorKit.jl`, and replace this to `infimum` everywhere, to make the tests work again.

**Comment by xiaoyu-dong** [comment]
Sure. I will do it :)

File: src/spaces/vectorspaces.jl (hunk 1 above)
Comment (xiaoyu-dong) [reply src/spaces/vectorspaces.jl]: Done!

File: src/spaces/vectorspaces.jl (hunk 2 above)
Comment (xiaoyu-dong) [reply src/spaces/vectorspaces.jl]: Done!

File: src/spaces/vectorspaces.jl (hunk 3 above)
Comment (xiaoyu-dong) [reply src/spaces/vectorspaces.jl]: Done!

**Comment by xiaoyu-dong** [comment]
I have changed all the infinum to infimum, and the tests work now.

**Comment by Jutho** [comment]
Hi @xiaoyu-dong , thanks again. Could it be that you did not push all the changes. There are still some "infinum"s remaining, e.g. in `TensorKit.jl/test/spaces.jl`, making tests to fail here. Maybe you only pushed the changes in `src/*`, but not in `test/*`.

**Comment by xiaoyu-dong** [comment]
> Hi @xiaoyu-dong , thanks again. Could it be that you did not push all the changes. There are still some "infinum"s remaining, e.g. in `TensorKit.jl/test/spaces.jl`, making tests to fail here. Maybe you only pushed the changes in `src/*`, but not in `test/*`.

oh, yes, sorry. I pushed it in the new commit.

**Comment by Jutho** [comment]
Not quite sure why the tests on Julia nightly failed, but everything seems fine on Julia 1.4 and 1.5, so I will merge and investigate further locally. Thanks for all of this @xiaoyu-dong . I still intend to write a paper about TensorKit.jl over summer, so you are very welcome to help.

**Pull request merged**